sqllogictest/src/runner.rs

```diff
@@ -1030,6 +1030,144 @@ pub fn update_record_with_output(
     }
 }
 
+/// Updates a test file with the output produced by a Database. It is an utility function wrapping
+/// [`update_test_file_with_runner`].
+///
+/// Specifically, it will create `"{filename}.temp"` to buffer the updated records and then override the
+/// original file with it.
+///
+/// Some other notes:
+/// - empty lines at the end of the file are cleaned.
+/// - `halt` and `include` are correctly handled.
+pub async fn update_test_file<D: AsyncDB>(
```

Comment:
Thank you for this @xxchan  -- I will attempt to integrate this upstream into DataFusion's sqllogictest runner hopefully this weekend

👍